### PR TITLE
Feature/hide blank items - WIP

### LINKED ISSUE #55: Add support for excluding empty values
* This should be a boolean option that defaults to false.

### PR DESCRIPTION
**Pull request by eric-kimbrel**
adds an option to hide blank items.  closes #55

### CODE REVIEW

**Comment by darthtrevino** [comment]
@eric-kimbrel Did you mean to close this without merging the changes?

**Comment by eric-kimbrel** [comment]
Yes, i found an issue i need to address.  When hiding blank items the value associated with that item still effects the scale used to render the other value bars which i think is not desired.   I'll get a fix for that worked up and open a new PR for it.

**Comment by darthtrevino** [comment]
Ah, I see; it's fine either way in this case, but in the future you don't have to delete PR's that aren't ready - just slap a "(WIP)" to the end of the title.  We can use PRs like that as kind of a zone for discussion.

**Comment by eric-kimbrel** [comment]
Ah, that makes sense.  I'll go ahead and reopen this one and mark it as WIP

**Comment by eric-kimbrel** [comment]
Because the slicer expects data items to be handed to it with the item width percentages already calculated the filtering of blank categories has to be done on on the VIsual side rather than in the slicer itself.  I've removed the logic added to AttributeSlicer and updated AttributeSlicerVisual with the needed logic to filter data before passing it on.

File: packages/attribute-slicer-powerbi/src/AttributeSlicerVisual.spec.ts [hunk 1]
```diff
@@ -382,6 +384,27 @@ describe("AttributeSlicerVisual", function () {
         expect(attributeSlicer.state.selectedItems.length).to.be.greaterThan(0);
     });
 
+    it("should not filter out blank categories by default", () => {
+        const { instance, attributeSlicer } = createInstance();
+        let fakeCats = ["CAT_1", "CAT_2", "", " "];
+        let update = createOptionsWithCategories(fakeCats, "SOME_CATEGORY_NAME");
+        instance.updateWithType(update, UpdateType.Data);
+
+        expect(attributeSlicer.data.map(function (n) { return n.match; })).to.be.deep.equal(fakeCats);
```
Comment (darthtrevino) [inline packages/attribute-slicer-powerbi/src/AttributeSlicerVisual.spec.ts]: Rewrite as lambda: `n => n.match`

File: packages/attribute-slicer-powerbi/src/AttributeSlicerVisual.spec.ts [hunk 2]
```diff
@@ -132,6 +133,7 @@ describe("AttributeSlicerVisual", function () {
                             Values: true,
                         },
                     }],
+                    objects: objects,
```
Comment (darthtrevino) [inline packages/attribute-slicer-powerbi/src/AttributeSlicerVisual.spec.ts]: Use es6 object shorthand: `objects,`

File: packages/attribute-slicer-powerbi/src/AttributeSlicerVisual.ts [hunk 3]
```diff
@@ -230,12 +229,14 @@ export default class AttributeSlicer extends VisualBase {
                         this.mySlicer.refresh();
                     }
 
-                    // The colors have changed, so we need to reload data
-                    if (!oldState.colors.equals(newState.colors)) {
+                    // The colors or hide items setting have changed, so we need to reload data
+                    if (!oldState.colors.equals(newState.colors) || oldState.hideEmptyItems !== newState.hideEmptyItems) {
```
Comment (darthtrevino) [inline packages/attribute-slicer-powerbi/src/AttributeSlicerVisual.ts]: extract to predicate `isReloadRequired`

**Pull request merged**